engine/src/main/java/org/terasology/rendering/dag/Node.java

```diff
@@ -28,6 +28,9 @@
 
     void process();
 
+    // TODO: invoked when Node is removed from RenderGraph
+    void dispose();
```

Comment:
Warning: for what I have seen in the codebase dispose() is called to give a chance to an object to do cleanup before it is destroyed and removed from memory. At this stage my thinking is that a node might continue to live even if it's not in the render graph. There are caveats to be discussed about this but I think they should be discussed within the context of the functionality to add/remove nodes to the graph and add/remove connections between them.